mllib/src/test/scala/org/apache/spark/ml/feature/RFormulaSuite.scala

```diff
@@ -290,4 +291,18 @@ class RFormulaSuite extends SparkFunSuite with MLlibTestSparkContext with Defaul
     val newModel = testDefaultReadWrite(model)
     checkModelData(model, newModel)
   }
+
+  test("should support all NumericType labels") {
```

Comment:
It'd work expect for the expected exception when dealing with a dataframe containing string labels because the label column gets indexed by RFormula's `fit`.
Consequently, an exception is thrown by `StringIndexer`.

What I could do is add a `validateSchema` to RFormula (called at the beginiing of the `fit` method) checking that the label column is of numeric type, then I could use `MLTestingUtils.checkNumericTypes`.

What do you think?